cbmc/proofs/mqtt_state.c

```diff
@@ -0,0 +1,556 @@
+					     void * pCredentialInfo,
+					     void * pConnection )
+{
+  // pCredentialInfo can be null
+  // create accepts NULL credentials when there is no TLS configuration.
+  __CPROVER_assert(pConnectionInfo != NULL,
+		   "IotNetworkInterfaceCreate pConnectionInfo is not NULL");
+  __CPROVER_assert(pConnection != NULL,
+		   "IotNetworkInterfaceCreate pConnection is not NULL");
+
+  // create the connection
+  *(void **)pConnection = allocate_IotNetworkConnection();
+
+  IotNetworkError_t error;
+  return error;
+}
+
+#define MAX_TRIES 2
+
+size_t IotNetworkInterfaceSend( void * pConnection,
+				const uint8_t * pMessage,
+				size_t messageLength )
+{
+  __CPROVER_assert(pConnection != NULL,
+		   "IotNetworkInterfaceSend pConnection is not NULL");
+  __CPROVER_assert(pMessage != NULL,
+		   "IotNetworkInterfaceSend pMessage is not NULL");
+
+  /****************************************************************
+   * The send method sends some portion of the message and returns the
+   * total number of bytes in the prefix sent so far.  The send method
+   * is used in a loop of the form
+   *
+   *   while ( send( conn, msg, len )  < len ) { ... }
+   *
+   * We need to bound the number of loop iterations, so we need to
+   * bound the number of times it takes for send to finish sending the
+   * message.  We use a static variable 'tries' to count the number of
+   * times send has tried to send the message, and we force send to
+   * finish the message after MAX_TRIES tries.
+   ****************************************************************/
+
+  // The number of tries to send the message before this invocation
+  static size_t tries;
+
+  // The number of bytes considered sent after this invocation
+  size_t size;
+
+  if (tries >= MAX_TRIES || size > messageLength )
+    {
+      tries = 0;
+      return messageLength;
+    }
+
+  tries++;
+  return size;
+}
+
+IotNetworkError_t IotNetworkInterfaceClose( void * pConnection )
+{
+  __CPROVER_assert(pConnection != NULL,
+		   "IotNetworkInterfaceClose pConnection is not NULL");
+
+  IotNetworkError_t error;
+  return error;
+}
+
+size_t IotNetworkInterfaceReceive( void * pConnection,
+				   uint8_t * pBuffer,
+				   size_t bytesRequested )
+{
+  __CPROVER_assert(pConnection,
+		   "IotNetworkInterfaceReceive pConnection is not NULL");
+  __CPROVER_assert(pBuffer,
+		   "IotNetworkInterfaceReceive pBuffer is not NULL");
+
+  // Fill the memory object pointed to by pBuffer with unconstrained
+  // data.  What follows a CBMC idiom to do that.
+  uint8_t byte;
+  __CPROVER_array_copy(pBuffer,&byte);
+
+  // Choose the number of bytes in pBuffer considered received.
+  size_t size;
+  __CPROVER_assume(size <= bytesRequested);
+
+  return size;
+}
+
+IotNetworkError_t IotNetworkInterfaceReceiveCallback( void * pConnection,
+					       IotNetworkReceiveCallback_t
+					         receiveCallback,
+					       void * pContext )
+{
```

Comment:
indent

Reply:
Fixed, I had pushed to the wrong repository.